index.js

```diff
@@ -32,7 +32,7 @@ module.exports = class WebpackCopyModulesPlugin {
   }
 
   apply(compiler) {
-    compiler.plugin('emit', this.handleEmit.bind(this));
+    compiler.hooks.emit.tapAsync('WebpackCopyModulesPlugin', this.handleEmit.bind(this));
```

Comment:
I suspect that `tapPromise` in conjunction with minor refactoring of `handleEmit` would be more appropriate

Reply:
Good call, done in 53aa6d9